### ISSUE TITLE
Supporting reactive-kafka

### ISSUE THREAD

**Issue by ghahramani** [spring-cloud/spring-cloud-stream#2166]
Hi,

After some discussion in the Gitter spring cloud stream channel, I realized to have a sequential pause/resume situation in reactive Kafka we need to use a back pressure mechanism that is not supported by web flux and Stream right now as we do not have any binder for `reactive-kafka` yet. I would like to raise this feature request for supporting `reactive-kafka` in spring cloud stream so we can easily use its functionalities.

**Comment by garyrussell** [comment]
@sabbyanandan @olegz This may not be trivial, but I think if we can pull it off, it should become the only way to support reactive applications in SCSt. 

Reactor apps on top of non-reactive binders (especially Kafka) is a fool's errand in my opinion.

There is also a `reactor-rabbitmq` project.

Perhaps we should spike this?

**Comment by olegz** [comment]
@garyrussell I completely agree and believe made it clear internally and externally. The same I believe would need to be made for Rabbit https://github.com/reactor/reactor-rabbitmq. This would also allow us to partition the binder API on reactive and imperative where reactive can still support the imperative model but not the other way around as it doesn't make any sense (as today)
